src/charm.py

```diff
@@ -370,6 +388,61 @@ def _pebble_layer(self):
         )
         return pebble_layer
 
+    @property
+    def self_probe(self):
+        """The self-monitoring blackbox probe."""
+        probe = {
+            'job_name': 'blackbox_http_2xx',
+            'params': {
+                'module': ['http_2xx']
+            },
+            'static_configs': [
+                {
+                    'targets': [self.external_url + "/api/v1/health/"],
+                    'labels': {'name': "cos-registration-server"}
+                }
+            ]
+        }
+        return [probe]
```

Comment:
How do you feel about decoupling config building from charm code? E.g.

```python
from config_builder import Config
# ...
probe = (
    Config()
    .job("blackbox_http_2xx")
    .params(module=["http_2xx"])
    .add_static_config(
        targets=[self.external_url + "/api/v1/health/"],
        labels={"name": "cos-registration-server"},
    ).build()
)
```

For now, this Config object, if you choose to extract it, could live in the blackbox charm lib, but ideally it should be written as if going to be handed over to the upstream project maintainer.